src/components/CallView/Grid/Grid.vue

```diff
@@ -268,6 +270,10 @@ export default {
 			// Timer for the videos bottom bar
 			showVideoOverlayTimer: null,
 			debounceMakeGrid: () => {},
+			videosTempPromoted: [],
+			unpromoteSpeakerTimer: {},
+			tilesWithVideoMask: [],
+			tilesWithAudioMask: [],
```

Comment:
If I was promoted with video enabled, then stopped speaking, and disable video, that would drop me after everyone who was promoted with audio only (and in reverse), without remaining the historical order
My suggestion is to unite both masks into a single queue, `First in - First out`, regardless of whether your video was on when you spoke, or not.

```suggestion
			promotedHistoryMask: [],
```

Shouldn't affect `getOrderedTiles()`, as it has two separate arrays (video / audio) to sort provided

Reply:
If we add the model to the mask during unpromotion, it will drop you in audios, first place.